BD-3392 Add LINE Login to User ID Reconciliation (#7926)
braze-inc/braze-docs: _docs/_user_guide/message_building_by_channel/line/line_setup.md

To combine a LINE ID with an existing Braze user profile, you can use the LINE login method.

### LINE Login

This method uses social media logins for reconciliation. When a user logs into your app, they're given the option to use [LINE Login](https://developers.line.biz/en/docs/line-login/overview/) to create a user account or log in.

{% alert note %}

> To acquire the correct LINE ID for each user, set up LINE Login under the same provider as your Braze integrated LINE official account or channel. 

Reviewer: ```suggestion
To acquire the correct LINE ID for each user, set up LINE Login under the same provider as your Braze-integrated LINE official account or channel. 
```

{% endalert %}

1. Go to the LINE Developer Console and [request permission to obtain the email addresses of users](https://developers.line.biz/en/docs/line-login/integrate-line-login/#applying-for-email-permission) who log into your app through LINE Login.

2. Follow the appropriate steps provided by LINE to implement LINE Login:<br><br>
  - [Web app directions](https://developers.line.biz/en/docs/line-login/integrate-line-login/)
  - [Native app directions](https://developers.line.biz/en/docs/line-login/secure-login-process/#using-openid-to-register-new-users)<br><br>Make sure to include `email` in the [scope set up](https://developers.line.biz/en/docs/line-login/integrate-line-login/#scopes) for verification requests. 

{: start="3"}
3. Use the [Verify ID token call](https://developers.line.biz/en/reference/line-login/#verify-id-token) to acquire the user’s email. 

4. Save the user’s LINE ID (`native_line_id`) to the user’s profile with a matching email in your database, or create a new user profile with the user’s email and LINE ID.

> 5. Send the new or updated user information to Braze using the [user/track]({{site.baseurl}}/api/endpoints/user_data/post_user_track#track-users/) endpoint, [CSV import]({{site.baseurl}}/user_guide/data_and_analytics/user_data_collection/user_import/#csv-import), or [Cloud Data Ingestion]({{site.baseurl}}/user_guide/data_and_analytics/cloud_ingestion/).

Reviewer: ```suggestion
5. Send the new or updated user information to Braze using the [`/user/track` endpoint]({{site.baseurl}}/api/endpoints/user_data/post_user_track#track-users/), [CSV import]({{site.baseurl}}/user_guide/data_and_analytics/user_data_collection/user_import/#csv-import), or [Cloud Data Ingestion]({{site.baseurl}}/user_guide/data_and_analytics/cloud_ingestion/).
```

#### Use cases

##### Existing follower uses LINE Login

> 1. Pre-existing state: An anonymous user was created during initial subscriber sync or after integration through a “follow” event.

Reviewer: ```suggestion
1. Pre-existing state: An anonymous user was created during initial subscriber sync or after integration through a “follow” event.
```
What does pre-existing state mean here? 🤔 Like scenario?

> 1. Pre-existing state: An anonymous user was created during initial subscriber sync or after integration through a “follow” event.
> 2. The user logs into your app using LINE Login.
> 3. LINE provides you the user’s email.
> 4. You send Braze the updated user (the existing user profile with that email to add the LINE ID) or you update the anonymous user with the email.

Author: ```suggestion
**Scenario:** An anonymous user was created during initial subscriber sync or after integration through a “follow” event.

1. The user logs into your app using LINE Login.
2. LINE provides you the user’s email.
3. You send Braze the updated user (the existing user profile with that email to add the LINE ID) or you update the anonymous user with the email.
```

##### New follower uses LINE Login

> 1. Pre-existing state: No user profile exists in Braze with user’s LINE ID.
> 2. The user logs into your app using LINE Login.
> 3. LINE provides you the user’s email.
> 4. You either:
>   - Update an existing user profile that has that email to also have the user’s LINE ID.
>   - Create a new user profile with the email and LINE ID.
> 5. When the user follows your LINE Official Account, Braze receives a follow event and updates the user’s subscription status to “Subscribed”.

Reviewer: ```suggestion
1. Pre-existing state: No user profile exists in Braze with the user’s LINE ID.
2. The user logs into your app using LINE Login.
3. LINE provides you with the user’s email.
4. You either:
  - Update an existing user profile with that email to also have the user’s LINE ID.
  - Create a new user profile with the email and LINE ID.
5. When the user follows your LINE Official Account, Braze receives a follow event and updates the user’s subscription status to “Subscribed”.
```

> 1. Pre-existing state: No user profile exists in Braze with the user’s LINE ID.
> 2. The user logs into your app using LINE Login.
> 3. LINE provides you with the user’s email.
> 4. You either:
>   - Update an existing user profile with that email to also have the user’s LINE ID.
>   - Create a new user profile with the email and LINE ID.
> 5. When the user follows your LINE Official Account, Braze receives a follow event and updates the user’s subscription status to “Subscribed”.

Author: ```suggestion
**Scenario:** No user profile exists in Braze with the user’s LINE ID.

1. The user logs into your app using LINE Login.
2. LINE provides you with the user’s email.
3. You either:
  - Update an existing user profile with that email to also have the user’s LINE ID.
  - Create a new user profile with the email and LINE ID.
4. When the user follows your LINE Official Account, Braze receives a follow event and updates the user’s subscription status to “Subscribed”.
```
